src/main/java/com/google/sps/servlets/FreeTimeServlet.java

```diff
@@ -0,0 +1,211 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.util.DateTime;
+import com.google.api.client.util.StringUtils;
+import com.google.api.services.calendar.model.CalendarListEntry;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.model.MessagePart;
+import com.google.common.io.BaseEncoding;
+import com.google.sps.data.PlanMailResponse;
+import com.google.sps.model.AuthenticatedHttpServlet;
+import com.google.sps.model.AuthenticationVerifier;
+import com.google.sps.model.CalendarClient;
+import com.google.sps.model.CalendarClientFactory;
+import com.google.sps.model.CalendarClientImpl;
+import com.google.sps.model.GmailClient;
+import com.google.sps.model.GmailClientFactory;
+import com.google.sps.model.GmailClientImpl;
+import com.google.sps.utility.DateInterval;
+import com.google.sps.utility.FreeTimeUtility;
+import com.google.sps.utility.JsonUtility;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.StringTokenizer;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.mail.MessagingException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** GET function responds JSON string containing potential events to read mail. */
+@WebServlet("/plan-mail")
+public class FreeTimeServlet extends AuthenticatedHttpServlet {
+  private final CalendarClientFactory calendarClientFactory;
+  private final GmailClientFactory gmailClientFactory;
+
+  /** Create servlet with default CalendarClient and Authentication Verifier implementations */
+  public FreeTimeServlet() {
+    calendarClientFactory = new CalendarClientImpl.Factory();
+    gmailClientFactory = new GmailClientImpl.Factory();
+  }
+
+  /**
+   * Create servlet with explicit implementations of CalendarClient and AuthenticationVerifier
+   *
+   * @param authenticationVerifier implementation of AuthenticationVerifier
+   * @param calendarClientFactory implementation of CalendarClientFactory
+   */
+  public FreeTimeServlet(
+      AuthenticationVerifier authenticationVerifier,
+      CalendarClientFactory calendarClientFactory,
+      GmailClientFactory gmailClientFactory) {
+    super(authenticationVerifier);
+    this.calendarClientFactory = calendarClientFactory;
+    this.gmailClientFactory = gmailClientFactory;
+  }
+
+  /**
+   * Returns string containing the some of the user's free time intervals, where events can be
+   * created
+   *
+   * @param request Http request from the client. Should contain idToken and accessToken
+   * @param response 403 if user is not authenticated, or Json string with the user's events
```

Comment:
Doc comment: technically speaking, this method should never be called is the response contains a 403 error. In this method, you  are always returning a Json string in the response.

Reply:
`Resolved`